source/grid/manifold.cc

```diff
@@ -131,6 +151,29 @@ get_new_point (const std::vector<Point<spacedim> > &surrounding_points,
 
 
 
+template <int dim, int spacedim>
+void
+Manifold<dim, spacedim>::
+add_new_points (const std::vector<Point<spacedim> > &surrounding_points,
+                const Table<2,double>               &weights,
+                std::vector<Point<spacedim> >       &new_points) const
+{
+  AssertDimension(surrounding_points.size(), weights.size(1));
+  Assert(&surrounding_points != &new_points,
+         ExcMessage("surrounding_points and new_points cannot be the same "
+                    "array"));
+
+  std::vector<double> local_weights(surrounding_points.size());
+  for (unsigned int row=0; row<weights.size(0); ++row)
+    {
+      std::memcpy(&local_weights[0], &weights(row,0),
+                  surrounding_points.size() * sizeof(double));
```

Comment:
This is making assumptions about the data layout of `weights`. Would it be significantly less efficient to let the compiler figure this out itself and convert a `for` loop into the equivalent statement?